data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJobV2.scala

```diff
@@ -82,11 +82,11 @@ object AssessmentMetricsJobV2 extends optional.Application with IJob with BaseRe
    * @return
    */
   def loadData(spark: SparkSession, settings: Map[String, String], url: String): DataFrame = {
-    spark
-      .read
-      .format(url)
-      .options(settings)
-      .load()
+    val schema = Encoders.product[UserData].schema
+    if(url.equals("org.apache.spark.sql.redis")) { spark.read.schema(schema).format(url).options(settings).load() }
+    else {
+      spark.read.format(url).options(settings).load()
+    }
```

Comment:
pass the schema as method parameter and register it only if it is not empty

Reply:
Done